### ISSUE TITLE
Closing a Dialog via custom Button throws IllegalArgumentException

### ISSUE THREAD

**Issue by xetra11** [JetBrains/compose-jb#199]
I have this logic on a close button inside the Dialog composition:
```kotlin
    Dialog(
        onDismissRequest = { DialogManager.closeDialog() // toggles rendering of dialog }
    ) {
        Column(
            Modifier.fillMaxSize().border(2.dp, Color.Blue),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.SpaceEvenly
        ) {
            CharacterCreateView()
            Button(onClick = { AppManager.focusedWindow?.close() }) {
                Text("Close")
            }
        }
    }

```

So when the "Close" button is clicked I expect the Dialog to close. It does however it can't be reopened. That seems to be cause of an exception thrown by the way I closed it:

```
Exception in thread "AWT-EventQueue-0" java.lang.IllegalArgumentException: Anchor refers to a group that was removed
	at androidx.compose.runtime.SlotTable.anchorIndex(SlotTable.kt:199)
	at androidx.compose.runtime.Anchor.toIndexFor(SlotTable.kt:466)
	at androidx.compose.runtime.Composer$CompositionReferenceImpl.getAmbientScope$runtime(Composer.kt:2542)
	at androidx.compose.runtime.Composer.startRoot(Composer.kt:549)
	at androidx.compose.runtime.Composer.recompose(Composer.kt:2094)
	at androidx.compose.runtime.Recomposer.performRecompose(Recomposer.kt:384)
	at androidx.compose.runtime.Recomposer.access$performRecompose(Recomposer.kt)
	at androidx.compose.runtime.Recomposer$recomposeAndApplyChanges$2$4.invoke(Recomposer.kt:318)
	at androidx.compose.runtime.Recomposer$recomposeAndApplyChanges$2$4.invoke(Recomposer.kt)
	at androidx.compose.runtime.dispatch.ActualDesktopKt$DefaultMonotonicFrameClock$2$1.withFrameNanos(ActualDesktop.kt:38)
	at androidx.compose.runtime.dispatch.ActualDesktopKt$DefaultMonotonicFrameClock$2$1$withFrameNanos$1.invokeSuspend(ActualDesktop.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTask.kt:227)
	at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:166)
	at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:362)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl(CancellableContinuationImpl.kt:396)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$default(CancellableContinuationImpl.kt:388)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeUndispatched(CancellableContinuationImpl.kt:484)
	at kotlinx.coroutines.swing.SwingDispatcher$scheduleResumeAfterDelay$timer$1.actionPerformed(SwingDispatcher.kt:33)
	at java.desktop/javax.swing.Timer.fireActionPerformed(Timer.java:317)
	at java.desktop/javax.swing.Timer$DoPostEvent.run(Timer.java:249)
	at java.desktop/java.awt.event.InvocationEvent.dispatch$$$capture(InvocationEvent.java:316)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```
The same happens when I use my Dialog rendering toggle `DialogManager.closeDialog()` (simply switching to an empty composition).
Is this a bug or do I just not have figured out the proper way to close a dialog via a button click event?

**Comment by xetra11** [comment]
Here is another minimal reproducable example:

```kotlin
fun main() {
    Window(title = "CK3 Mod Workbench") {
        MaterialTheme() {
            val currentDialog = remember { mutableStateOf("noDialog") }
            if (currentDialog.value == "characterDialog") Dialog(
                onDismissRequest = { currentDialog.value = "noDialog" }
            ) {
                Button(onClick = { currentDialog.value = "noDialog" }) {
                    Text("Close Dialog")
                }
            }
            Button(onClick = { currentDialog.value = "characterDialog" }
            ) { Text("Open Dialog") }
        }
    }
}
```

**Comment by Rsedaikin** [comment]
Fixed in version `0.3.0-build138`.

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6495">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
